LIBVmotor0.11/LIBVmotor0.11.ino

```diff
@@ -1,172 +1,249 @@
+ */
+ 
+#include <EEPROM.h>
+#include <Wire.h>
+#include <Encoder.h>
+#include <simplePID.h>
+#include <BTS7960.h>
+#include <button.h>
+
+#define STATION_I2C_ADD 8 //I2C slave station Address
+
+//VAL: EPPROM Address
+const int TV_ADD= 10, TI_ADD=11, IE_ADD=12;
+
+//P-IN: Hall switch
+#define HALL1_PIN 11
+#define HALL2_PIN 10
+
+//P-IN: Button
+#define BREATH_PIN 12
+
+//P-IN: Encoder
+#define chA 2   //Motor's encoder channel A
+#define chB 3   //Motor's encoder channel B
+
+//P-IN: Analog sensor
+#define PRES1_PIN A2
+#define PRES2_PIN A3
+
+//P-OUT: Motor driver, BTS7960
+#define LPWM 5
+#define RPWM 6
+
+//VAL: Motor's spec
+#define mGear 51.0  //Motor's gear ratio
+#define mPpr 11.0   //Number of tick per one revolution capture by the encoder
+#define mPulse 4.0  //Always 4, the encoder library return 4 in 1 tick from the encoder
+
+//VAL: PID Parameters
+#define Kp 140.0    //Proportional gain
+#define Ki 120.0    //Integral gain
+#define ZERO 0.0    //Zero value
+
+//VAR: PID Variable 
+long t, xt, dt;                   //timer
+long En, xEn;                     //Store the encoder value
+float goPos=0.0, aPos=0.0, dPos;  //Store the Position
+float goSpeed=0.0, aSpeed=0.0;    //Target Speed, current Speed
+int cmd;                          //digital command from PID, the PWM value
+
+//Setting Variable
+#define RPB 0.62           //Number of round to rotate to one breath inhale
+int TV, TI, IE;
+float TV_MIN=0, TV_MAX=680;
+float TV2RPB=0.0;
+//bool pushIn=false;                  //define action of push IN or release the bag
+int timeIn, timeOut, timeRest;                //time to push in and push out in millisecond
+float speedIn, speedOut, speedMov;  //speed to push in and push out of the motor, in round per second (rps)
+unsigned long timer=0;              //to store the timer of running, millis(), millisecond
+
+//VAR: Ven stat
+int venStat=0;
+const int pushIN=0;
+const int pushOUT=1;
+const int pushRest=2;
+
+//VAR: Breath Button
+bool breathing = false;
+bool brth_hold_stat = false;
+const int brth_hold_time = 3000;
+
+simplePID PiD(Kp, Ki);      //Kp, Ki, use only PI, name it to "PiD"
+ramp Ramp(3.0, 2.0, 0.005); //Acceleration, Max Speed, Tolerance, name it to "Ramp"
+Encoder Enc(chA, chB);      //Encoder pins, name it to "Enc"
+BTS7960 motor(LPWM, RPWM);
+
+button BREATH_BUT(BREATH_PIN);
+//button HALL1(HALL1_PIN);
+//button HALL2(HALL2_PIN);
+
+void setup() {
+  Serial.begin(9600);
+
+  //Initialize the digital input
+  BREATH_BUT.begin(true);
+  pinMode(HALL1_PIN, INPUT);
+  //HALL1.begin(false);
+  //HALL2.begin(false);
+
+  //I2C Setup
+  Wire.begin(STATION_I2C_ADD);                // join i2c bus with address #8
+  Wire.onReceive(receiveEvent);       // I2C receive data event
+
+  
+  EEPROM_read(); //Get Setting from EEPROM
+  BPM_Timing();  //Calc breath timing
+
+  
+  motor.init();  //Motor Setup
+
+  //Initialize the Arms
+  while(true){
+    motor.setPWM(-200);
```

Comment:
Where does this -200 come from? It's a speed for homing, right? Give it a name and put it in a #define so that we can adjust it without digging into the code 😄